src/cli/ionic-config.ts

```diff
@@ -21,6 +21,10 @@ export async function readConfig(): Promise<TelemetryConfig> {
     };
 
     await writeConfig(config);
+  } else if (!!config && !config['tokens.telemetry'].match(UUID_REGEX)) {
+    const newUuid = uuidv4();
+    await writeConfig(Object.assign(config, { 'tokens.telemetry': newUuid }));
```

Comment:
FYI/no action required - from a TypeScript typing perspective, there is a subtle difference between using `Object.assign` and the spread operator. `Object.assign` is typed such that the return type is the intersection of the two args:
```typescript
assign<T, U>(target: T, source: U): T & U;
```
if we were to mess up the name of the key in `U`, the compiler wouldn't catch the error:
```typescript
// compiles just fine
await writeConfig(Object.assign(config, { 'tokens.teleasdf': newUuid }));
```

whereas spreading the object will catch an error
```typescript
// errors! we can't assign to type `TelemetryConfig`
await writeConfig({ ...config, 'tokens.teleasdf': newUuid });
```

There are some subtle differences between the two (see ['Differences Versus Object.assign()'](https://thecodebarbarian.com/object-assign-vs-object-spread.html)) but I don't think they apply here

Reply:
Thank you. I did this because I wanted to modify the object in place, but the TypeScript side effect is news to me... Would you like to add `prefer-object-spread: error` to our ESLint rules? Open to either option.

Reply:
Maybe WRT `prefer-object-spread`, I need to start having that conversation with Lars this week about long term linting & formatting plans